### PR TITLE
Revise tech docs in README to favour GOV.UK Docker

### PR DESCRIPTION
**Pull request by thomasleese**
This makes several changes to refactor and revise the README to better match our other repos. The end result follows the pattern established in https://github.com/alphagov/content-publisher/pull/2257. See the commit messages for more details.

[Trello Card](https://trello.com/c/Psf7EzPB/204-move-away-from-startupsh)

### CODE REVIEW

**Review by brucebolt — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -11,40 +11,19 @@ Service Manual Publisher is used for publishing and maintaining https://www.gov.
...
 
-PostgreSQL-backed Rails publishing application for internal use, with no public facing aspect.
+This is a Ruby on Rails app, and should follow [our Rails app conventions](https://docs.publishing.service.gov.uk/manual/conventions-for-rails-applications.html).
 
 There is some [disparity between the content as it appears in the database, and the content as it appears in the publishing api](doc/arch/002-disparity-between-database-and-publishing-api.md). This will need to be addressed if we switch to using the Publishing API as our main data store in the future.
 
 The Service manual's guide pages have a nested URL structure which is unusual for GOV.UK. The pros and cons to the nested URL structure are [retrospectively documented](doc/arch/001-nested-url-structure.md).
 
-### Development
+You can use the [GOV.UK Docker environment](https://github.com/alphagov/govuk-docker) to run the application and its tests with all the necessary dependencies. Follow [the usage instructions](https://github.com/alphagov/govuk-docker#usage) to get started.
```
Comment (benthorner) [inline README.md:20]: We normally add a final line here: "Use GOV.UK Docker to run any commands that follow."

File: README.md [hunk 2]
```diff
@@ -11,40 +11,19 @@ Service Manual Publisher is used for publishing and maintaining https://www.gov.
 
 ## Technical documentation
 
-PostgreSQL-backed Rails publishing application for internal use, with no public facing aspect.
+This is a Ruby on Rails app, and should follow [our Rails app conventions](https://docs.publishing.service.gov.uk/manual/conventions-for-rails-applications.html).
 
 There is some [disparity between the content as it appears in the database, and the content as it appears in the publishing api](doc/arch/002-disparity-between-database-and-publishing-api.md). This will need to be addressed if we switch to using the Publishing API as our main data store in the future.
```
Comment (benthorner) [inline README.md:16]: We could move these to a later "Further documentation" section, and maybe help out @ChrisBAshton by renaming `doc/ -> docs`.

**Review by benthorner — APPROVED** [review]
Nice, so much shorter! I've left a couple of more minor comments based on some of the quick-wins we've done in other PRs.

This app is one of the few that doesn't actually have [web stack support](https://github.com/alphagov/govuk-docker/blob/master/docs/compatibility.md) in GOV.UK Docker. I don't think that's a major issue, though: the basic config is there, it's easy to add web stack support, and I might try and look at doing that myself (just for completeness).

File: README.md (hunk 2 above)
Comment (thomasleese) [reply README.md]: Good point, solved in 78bbf1a and d84669f.

File: README.md (hunk 1 above)
Comment (thomasleese) [reply README.md:16]: I've amended the commit in dc2bafa to fix this.

**Pull request merged**